src/lib.rs

```diff
@@ -2341,6 +2341,26 @@ impl Connection {
 
                     self.peer_transport_params = peer_params;
 
+                    // Update max_data of current sender streams with some data
+                    // already written. This can happen when there is a
+                    // Handshake packet loss but received actual request already,
+                    // so the stream (usually id 0) is already opened and have some
+                    // data to write, but actual packetization is prevented because of
+                    // max_tx_data = 0.
+                    // What we do here is update max_data if max_data=0
+                    // In normal situation (handshake is complete before getting
+                    // request) it will not do anything.
+                    for (id, stream) in self.streams.iter_mut() {
+                        if stream.send.get_max_data() == 0 {
```

Comment:
It is possible to condense this to:

```
for (id, stream) in self.streams.iter_mut().filter(|(_, s)| s.send.get_max_data() == 0) {
 ...
}
```

although you'll need to make `get_max_data` non-mutable

Reply:
Thanks. I tried it before and wondering why build failed...